packages/jellyfish-api-core/__tests__/category/vault/estimateVault.test.ts

```diff
@@ -0,0 +1,169 @@
+    // Setup non collateral token
+    await tGroup.get(0).token.create({ symbol: 'DOGE', collateralAddress })
+    await tGroup.get(0).generate(1)
+    await tGroup.get(0).token.mint({ symbol: 'DOGE', amount: 50000 })
+    await tGroup.get(0).generate(1)
+
+    // oracle setup
+    const addr = await tGroup.get(0).generateAddress()
+    const priceFeeds = [
+      { token: 'DFI', currency: 'USD' },
+      { token: 'BTC', currency: 'USD' },
+      { token: 'DOGE', currency: 'USD' },
+      { token: 'TSLA', currency: 'USD' },
+      { token: 'MSFT', currency: 'USD' }
+    ]
+    const oracleId = await tGroup.get(0).rpc.oracle.appointOracle(addr, priceFeeds, { weightage: 1 })
+    await tGroup.get(0).generate(1)
+    const oracleTickTimestamp = Math.floor(new Date().getTime() / 1000)
+    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, oracleTickTimestamp, {
+      prices: [
+        { tokenAmount: '1@DFI', currency: 'USD' },
+        { tokenAmount: '0.01@DOGE', currency: 'USD' },
+        { tokenAmount: '10000@BTC', currency: 'USD' },
+        { tokenAmount: '2@TSLA', currency: 'USD' },
+        { tokenAmount: '5@MSFT', currency: 'USD' }
+      ]
+    })
+    await tGroup.get(0).generate(1)
+
+    // collateral token
+    await tGroup.get(0).rpc.loan.setCollateralToken({
+      token: 'DFI',
+      factor: new BigNumber(1),
+      fixedIntervalPriceId: 'DFI/USD'
+    })
+    await tGroup.get(0).generate(1)
+
+    // collateral token
+    await tGroup.get(0).rpc.loan.setCollateralToken({
+      token: 'BTC',
+      factor: new BigNumber(0.5),
+      fixedIntervalPriceId: 'BTC/USD'
+    })
+    await tGroup.get(0).generate(1)
+
+    // loan token
+    await tGroup.get(0).rpc.loan.setLoanToken({
+      symbol: 'TSLA',
+      fixedIntervalPriceId: 'TSLA/USD'
+    })
+    await tGroup.get(0).generate(1)
+    await tGroup.get(0).rpc.loan.setLoanToken({
+      symbol: 'MSFT',
+      fixedIntervalPriceId: 'MSFT/USD'
+    })
+    await tGroup.get(0).generate(1)
+
+    // loan scheme set up
+    await tGroup.get(0).rpc.loan.createLoanScheme({
+      minColRatio: 150,
+      interestRate: new BigNumber(3),
+      id: 'scheme'
+    })
+    await tGroup.get(0).generate(1)
+
+    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, Math.floor(new Date().getTime() / 1000), {
+      prices: [{ tokenAmount: '2@TSLA', currency: 'USD' }, { tokenAmount: '5@MSFT', currency: 'USD' }]
+    })
+    await tGroup.get(0).generate(1)
+
+    await tGroup.waitForSync()
+    await tGroup.get(0).container.waitForPriceValid('TSLA/USD')
+    await tGroup.get(0).container.waitForPriceValid('MSFT/USD')
+
+    return { collateralAddress, oracleId }
+  }
+
+  beforeAll(async () => {
+    await tGroup.start()
+    await tGroup.get(0).container.waitForWalletCoinbaseMaturity()
+    await setup(tGroup)
+  })
+
+  afterAll(async () => {
+    await tGroup.stop()
+  })
+
+  it('should fail if collateralAmounts array contains non collateral token', async () => {
```

Comment:
I am thinking if you should set up multi-nodes for this test. I don't see any of the test items using get(0)